cli/tools/registry/pm.rs

```diff
@@ -28,23 +29,43 @@ use crate::file_fetcher::FileFetcher;
 use crate::jsr::JsrFetchResolver;
 use crate::npm::NpmFetchResolver;
 
-enum ConfigFile {
-  Deno(deno_config::ConfigFile),
+enum DenoConfigFormat {
+  Json,
+  Jsonc,
+}
+
+impl DenoConfigFormat {
+  fn from_specifier(spec: &ModuleSpecifier) -> Result<Self, AnyError> {
+    let file_name = spec
+      .path_segments()
+      .ok_or_else(|| anyhow!("Empty path in deno config specifier: {spec}"))?
+      .last()
+      .unwrap();
+    match file_name {
+      "deno.json" => Ok(Self::Json),
+      "deno.jsonc" => Ok(Self::Jsonc),
+      _ => bail!("Unsupported deno config file: {file_name}"),
+    }
+  }
+}
+
+enum DenoOrPackageJson {
+  Deno(deno_config::ConfigFile, DenoConfigFormat),
   Npm(deno_node::PackageJson, Option<FmtOptionsConfig>),
 }
 
-impl ConfigFile {
-  fn specifier(&self) -> ModuleSpecifier {
+impl DenoOrPackageJson {
+  fn specifier(&self) -> Cow<ModuleSpecifier> {
     match self {
-      Self::Deno(d) => d.specifier.clone(),
-      Self::Npm(n, ..) => n.specifier(),
+      Self::Deno(d, ..) => Cow::Borrowed(&d.specifier),
+      Self::Npm(n, ..) => Cow::Owned(n.specifier()),
```

Comment:
Oh, sorry, I didn't notice this was returning an owned value and wouldn't have bothered with my last comment, but this is fine.